homeassistant/components/climate/__init__.py

```diff
@@ -364,6 +361,12 @@ def state_attributes(self):
             ATTR_TEMPERATURE:
             self._convert_for_display(self.target_temperature),
         }
+        target_temp_high = self.target_temperature_high
+        if target_temp_high is not None:
+            data[ATTR_TARGET_TEMP_HIGH] = self._convert_for_display(
+                self.target_temperature_high)
+            data[ATTR_TARGET_TEMP_LOW] = self._convert_for_display(
```

Comment:
with both high and low set as `vol.Optional` - this may possibly bomb out, especially if the `set_temperature` implementation doesn't do a None check and/or an initial value.